src/helpers/wallet.helper.js

```diff
@@ -0,0 +1,108 @@
+/**
+ * Copyright (c) Hathor Labs and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import { HathorWallet, walletUtils, Connection, errors } from '@hathor/wallet-lib';
+import config from '../config';
+
+export class WalletStartError extends Error {}
+
+export function getReadonlyWallet({
+  xpub,
+  multisigData = null,
+  preCalculatedAddresses = [],
+}) {
+  const connection = new Connection({
+    network: config.network,
+    servers: [config.server],
+    connectionTimeout: config.connectionTimeout,
+  });
+
+  // Previous versions of the lib would have password and pin default as '123'
+  // We currently need something to be defined, otherwise we get an error when starting the wallet
+  const walletConfig = {
+    xpub,
+    connection,
+    password: '123',
+    pinCode: '123',
+    multisig: multisigData,
+  };
+
+  // tokenUid is optional but if not passed as parameter the wallet will use HTR
+  if (config.tokenUid) {
+    walletConfig.tokenUid = config.tokenUid;
+  }
+
+  // Wallet addresses pre-calculation, usually for speeding up tests
+  if (preCalculatedAddresses && preCalculatedAddresses.length) {
+    console.log(`Received pre-calculated addresses`, preCalculatedAddresses);
+    walletConfig.preCalculatedAddresses = preCalculatedAddresses;
+  }
```

Comment:
All those lines are the same for both methods. Why don't you keep this code in the `/start` controller and send here after it's handled?

Reply:
These are on the wallet-helper because it directly affects the `walletConfig` which is internal to these method.

The `preCalculatedAddresses` are extracted on the `/start` handler here we just add them to the `walletConfig` if they exist.

The only way to move this to the `/start` handler would be to make these methods only build the `walletConfig` and we could update it on the `/start` handler before initiating the wallet.

Reply:
I just don't like the duplicated code. If you move this the same way you moved the `connection` code we would prevent this duplicated code.